main.go

```diff
@@ -64,24 +65,46 @@ func InfoHandler(contentAPI, needAPI, performanceAPI string, config *Config) fun
 			return
 		}
 
-		needStart := time.Now()
-		for _, needID := range artefact.Details.NeedIDs {
-			need, err := need_api.FetchNeed(needAPI, config.BearerTokenNeedAPI, needID)
+		var waitGroup sync.WaitGroup
+		errorChannel := make(chan string)
+
+		waitGroup.Add(1)
+		go func() {
+			defer waitGroup.Done()
+
+			needStart := time.Now()
+			for _, needID := range artefact.Details.NeedIDs {
+				need, err := need_api.FetchNeed(needAPI, config.BearerTokenNeedAPI, needID)
+				if err != nil {
+					errorChannel <- "Need:" + err.Error()
+					return
+				}
+				needs = append(needs, need)
+			}
+			statsDTiming("needs", needStart, time.Now())
+		}()
+
+		waitGroup.Add(1)
+		var performance *performance_platform.Statistics
+		go func() {
+			defer waitGroup.Done()
+
+			performanceStart := time.Now()
+			ppClient := performanceclient.NewDataClient(performanceAPI, logging)
+			is_multipart := (len(artefact.Details.Parts) != 0) || (artefact.Format == "smart-answer")
+			performance, err = performance_platform.SlugStatistics(ppClient, slug, is_multipart)
+			statsDTiming("performance", performanceStart, time.Now())
 			if err != nil {
-				renderError(w, http.StatusInternalServerError, "Need: "+err.Error())
+				errorChannel <- "Performance:" + err.Error()
 				return
 			}
-			needs = append(needs, need)
-		}
-		statsDTiming("needs", needStart, time.Now())
+		}()
 
-		performanceStart := time.Now()
-		ppClient := performanceclient.NewDataClient(performanceAPI, logging)
-		is_multipart := (len(artefact.Details.Parts) != 0) || (artefact.Format == "smart-answer")
-		performance, err := performance_platform.SlugStatistics(ppClient, slug, is_multipart)
-		statsDTiming("performance", performanceStart, time.Now())
-		if err != nil {
-			renderError(w, http.StatusInternalServerError, "Performance: "+err.Error())
+		waitGroup.Wait()
+
+		defer close(errorChannel)
```

Comment:
I don't think this is necessary. Channels are garbage collected like everything else when they're no longer referenced. It's not necessary to close them first. Closing channels is more typically used to signal to readers that nothing further will be sent on the channel.